modules/runner/src/main/java/org/apache/ignite/internal/configuration/storage/DistributedConfigurationStorage.java

```diff
@@ -111,88 +120,95 @@ public DistributedConfigurationStorage(MetaStorageManager metaStorageMgr, VaultM
         this.vaultMgr = vaultMgr;
     }
 
+    @Override
+    public void close() throws Exception {
+        IgniteUtils.shutdownAndAwaitTermination(threadPool, 10, TimeUnit.SECONDS);
```

Comment:
`InFlightFutures` can be used to track and, on stop, cancel the futures that will never be resolved due to not having a chance to be executed